Healex.HL7v2Anonymizer/Services/Anonymizer.cs

```diff
@@ -26,89 +26,20 @@ public class Anonymizer {
 
         public void Anonymize(Message message)
         {
-            for (var index = 0; index < message.SegmentCount; index++) {
-                var segment = message.Segments()[index]; 
-                var substitution = options.Segments.FirstOrDefault(s => s.Segment == segment.Name);
-                if (substitution == null) continue;
-
-                var rules = ExpandReplacementRulesForRepeatedFields(segment, substitution.Replacements);
-                
-                foreach (var replacement in rules)
-                {
-                    var value = Replacement(replacement, message);
-                    TryReplaceValue(message, replacement.Path, value);
-                }
-            }
-        }
-
-
-        #endregion
-
-        #region Auxiliar Methods
-
-        private static List<Replacement> ExpandReplacementRulesForRepeatedFields(Segment segment, Replacement[] originalRules)
-        {
-            var rules = new List<Replacement>();
-            rules.AddRange(originalRules);
-                
-            foreach (var replacement in originalRules)
+            foreach (var segment in message.Segments())
             {
-                var indices = GetIndices(replacement.Path);
-                var fieldIndex = indices[0];
-                if (segment.Fields(fieldIndex) == null || !segment.Fields(fieldIndex).HasRepetitions) continue;
-                    
-                for (var i=1; i < segment.Fields(fieldIndex).Repetitions().Count; i++)
+                var rules = options.Segments.Where(replacement => replacement.Segment == segment.Name)
+                    .Select(replacement => replacement.Replacements);
```

Comment:
Why `Where` instead of `FirstOrDefault`? Wouldn't this support dealing with duplicate sections in the config? I mean, the user shouldn't configure the anonymizer like that. But what is expected to happen if a user configures like the PID section 2 times with the 1st section containing a HASH replacement, while the 2nd section contains a constant value? I would argue that iterating over every matching rule might introduce ambigous behaviour (also because the order in a JSON list is not guaranteed).